lib/parsec/OrderedDict.py

```diff
@@ -99,3 +99,29 @@ def __contains__(self, key):
     def __nonzero__(self):
         """Include any default keys in the nonzero calculation."""
         return bool(self.keys())
+
+    def prepend(self, key, value, dict_setitem=dict.__setitem__):
+        """Prepend new item in the ordered dict.
+
+        https://stackoverflow.com/questions/16664874/
+           how-can-i-add-an-element-at-the-top-of-an-ordereddict-in-python
+
+        Tested with Python 2.7 collections.OrderedDict and our bundled
+        ActiveState implementation for Python 2.6.
+
+        """
+        root = self._OrderedDict__root
+        first = root[1]
+
+        if key in self:
+            link = self._OrderedDict__map[key]
+            link_prev, link_next, _ = link
+            link_prev[1] = link_next
+            link_next[0] = link_prev
+            link[0] = root
+            link[1] = first
+            root[1] = first[0] = link
+        else:
+            root[1] = first[0] = self._OrderedDict__map[key] = [
+                root, first, key]
+            dict_setitem(self, key, value)
```

Comment:
Is this line at the wrong indent level? At the moment if you prepend an existing key its value will not get updated. e.g:

```python
>>> dictionary = OrderedDictWithDefaults()
>>> dictionary['foo'] = 1
>>> dictionary['bar'] = 2
>>> dictionary
OrderedDictWithDefaults([('foo', 1), ('bar', 2)])

>>> dictionary.prepend('baz', 0)
>>> dictionary
OrderedDictWithDefaults([('baz', 0), ('foo', 1), ('bar', 2)])

>>> dictionary.prepend('foo', 'Fool')  # Value won't get changed!
>>> dictionary
OrderedDictWithDefaults([('foo', 1), ('baz', 0), ('bar', 2)])

```

Reply:
Good spotting.